packages/core-blockchain/src/processor/block-processor.ts

```diff
@@ -112,4 +117,44 @@ export class BlockProcessor {
 
         return false;
     }
+
+    /**
+     * Check if a block's transactions violate nonce order.
+     * For a given sender, all v1 transactions must precede all v2 transactions.
+     * In addition, for a given sender, v2 transactions must have strictly
+     * increasing nonce without gaps.
+     */
+    private blockContainsOutOfOrderNonce(block: Interfaces.IBlock): boolean {
+        const nonceBySender = {};
+
+        for (const transaction of block.transactions) {
+            const data = transaction.data;
+
+            if (data.version < 2) {
+                continue;
+            }
+
+            const sender: string = data.senderPublicKey;
+
+            if (nonceBySender[sender] === undefined) {
+                nonceBySender[sender] = this.blockchain.database.walletManager.getNonce(sender);
+            }
+
+            const currentTransactionNonce = Utils.BigNumber.make(data.nonce);
```

Comment:
`data.nonce` is already a bignumber so no need to make a new one.

Reply:
Simplified in 0d55b6d8b.